File: Arcmage.Web/src/app/models/deck.ts
import { Base } from "./base";
import { DeckCard } from "./deck-card";
import { Status } from "./status";

export class Deck extends Base {
  id: number;
  name: string;
  zip: string;
  txt: string;
  exportTiles: boolean;
  generatePdf: boolean;
  isAvailable: boolean;
  isEditable: boolean;
  deckCards: DeckCard[];
  totalCards: number;
  isGenerated: boolean;
  status: Status;
}

File: Arcmage.Model/Deck.cs
using System.Collections.Generic;

namespace Arcmage.Model
{
    public class Deck : Base
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Zip { get; set; }

        public string Txt { get; set; }

        public bool ExportTiles { get; set; }

        public bool GeneratePdf { get; set; }

        public bool IsAvailable { get; set; }

        public bool IsEditable { get; set; }

        public bool IsGenerated { get; set; }

        public Status Status { get; set; }

        public List<DeckCard> DeckCards { get; set; }

        public int TotalCards { get; set; }

        public Deck()
        {
            DeckCards = new List<DeckCard>();
        }

    }
}
